SWLOR.Game.Server/Feature/ShipModuleDefinition/HypermatterInjectorModuleDefinition.cs

```diff
@@ -51,7 +51,7 @@ private void HypermatterInjector(string tag, string name, string shortName, stri
                 })
                 .ActivatedAction((activator, activatorShipStatus, target, targetShipStatus, moduleBonus) =>
                 {
-                    if (!GetIsObjectValid(target))
+                    if (!GetIsObjectValid(target) || GetIsEnemy(target, activator))
```

Comment:
What's the use case here?

Reply:
It is what makes the game automatically fire the device on self if target is hostile. I don't question how it works. it just does.

Reply:
If other players are targeted in space I don't think it automatically hostiles them. In that case I don't think this will work as intended when targeting players.

I'm questioning whether the if statement is needed at all. If the expectation is that no matter what your target is, the hypermatter injector should be used then I'd say remove the if statement altogether. It seems unnecessary based on what I know.

Reply:
I mean the shield and hull use the same code. I just based it off those and they work fine. This is just checking to see if the target is hostile and if so it fires on self. If its another player or friendly it fires on them.

Reply:
Ok I see now. It's fine to leave this